adr/ADR-20.md

```diff
@@ -216,54 +290,73 @@ is recommended but optional if it does not make sense for the client language.
 ```go
 type ObjectStore interface {
     // Put will place the contents from the reader into a new object.
-    Put(obj *ObjectMeta, reader io.Reader, opts ...ObjectOpt) (*ObjectInfo, error)
+    // It is an error to Put when ObjectMeta contains a Link. Use AddLink or AddBucketLink
```

Comment:
Not sure why we are copying the go interface here. Any functionality that beyond put/get that returns byte is specific to the client and can be optional implementation.

Instead, put the basic interface for the object store, put, get, link, delete, watch, seal, status, and then put a link to the go code.

Reply:
I have addressed these comments.